iothub/device/devdoc/API Design.md

```diff
@@ -0,0 +1,490 @@
+## Plug and Play convention compatible APIs
+
+### Properties
+
+```csharp
+/// <summary>
+/// Retrieve the device properties.
+/// </summary>
+/// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+/// <returns>The device properties.</returns>
+public Task<Properties> GetPropertiesAsync(CancellationToken cancellationToken = default);
+
+/// <summary>
+/// Update a single property.
+/// </summary>
+/// <param name="propertyName">Property name.</param>
+/// <param name="propertyValue">Property value.</param>
+/// <param name="propertyConvention">A convention handler that defines serializer to use for the properties.</param>
+/// <param name="componentName">The component name this property belongs to.</param>
+/// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+public Task UpdatePropertyAsync(string propertyName, object propertyValue, PropertyConvention propertyConvention, string componentName = default, CancellationToken cancellationToken = default);
```

Comment:
used PropertyConvention with virtual methods for serialization/ deserialization (vs interfaces), since it is easier to override default behavior (pending internal team discussion though)